cli/azd/pkg/tools/github/github.go

```diff
@@ -187,15 +187,15 @@ type AuthStatus struct {
 func (cli *ghCli) GetAuthStatus(ctx context.Context, hostname string) (AuthStatus, error) {
 	runArgs := cli.newRunArgs("auth", "status", "--hostname", hostname)
 	res, err := cli.commandRunner.Run(ctx, runArgs)
-	if res.ExitCode == 0 {
+	if err == nil && res.ExitCode == 0 {
```

Comment:
This fixes a bug where if `Run` failed due to a non-exit-code error, A default `RunResult` is returned with `ExitCode` set to 0, and this logic would assume that meant the command successfully exited.

Reply:
You're right that it is indeed redundant. I also made an invalid assumption earlier, when I said `ExitCode` is 0 when the program exits early. It is actually set to `-1` by ProcessState.ExitCode(). The existing code did work as expected, but I'll update it to be more inline with err checks.